### PR TITLE
feature(cs_driver): Check is shared awarnes or not driver is used

### PR DESCRIPTION
**Pull request by aleksbykov**
Using prometheus query and parsing c-s log file check
whether c-s use shared-aware driver or not and add info
to summary email

Trello: https://trello.com/c/VrpxJLr4

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by aleksbykov** [comment]
Screenshot from 2021-08-13 17-27-00

File: sdcm/tester.py [hunk 1]
```diff
@@ -2924,6 +2924,16 @@ def all_nodes_scylla_shards(self):
 
         return all_nodes_shards
 
+    def is_shard_aware_driver_used(self):
```
Comment (dkropachev) [inline sdcm/tester.py:2927]: Is this used anywhere ?

**Review by dkropachev — CHANGES_REQUESTED** [review]
Whell there are two problems:
1. What are we going to do with non java drivers ? Right now all of the are going to be marked as non shardaware, which is not true.
2. Even if we get this message it does not mean that connection is running in shard-aware mode

Maybe we could just analyze scylla_storage_proxy_replica_cross_shard_ops and draw conclusion from there ?

File: sdcm/db_stats.py [hunk 2]
```diff
@@ -317,6 +319,16 @@ def get_scylla_scheduler_shares_per_sla(self, start_time, end_time, node_ip):  #
             res[item['metric']['group']] = {int(i[1]) for i in item['values']}
         return res
 
+    def get_scylla_scylla_storage_proxy_replica_cross_shard_ops(self, start_time, end_time):
```
Comment (fgelcer) [inline sdcm/db_stats.py:322]:
```suggestion
    def get_scylla_storage_proxy_replica_cross_shard_ops(self, start_time, end_time):
```

File: sdcm/report_templates/results_base.html [hunk 3]
```diff
@@ -205,6 +205,9 @@ <h3>Test details</h3>
             {% if gemini_version %}
             <li>Gemini version: {{ gemini_version }}</li>
             {% endif %}
+            {% if shard_awareness_driver %}
+                <li> Cassandra-stress uses shared-aware driver</li>
```
Comment (fgelcer) [inline sdcm/report_templates/results_base.html:209]: Do we have access to versions of Scylla shard aware drivers?

**Comment by aleksbykov** [comment]
> Whell there are two problems:
> 
> 1. What are we going to do with non java drivers ? Right now all of the are going to be marked as non shardaware, which is not true.

Yes, and it is a problem. 


> 2. Even if we get this message it does not mean that connection is running in shard-aware mode
> 
> Maybe we could just analyze scylla_storage_proxy_replica_cross_shard_ops and draw conclusion from there ?

The problem with this metric, that it doesn't contain name of stress tool ), so even if run 2 toosl: c-s with shard-aware and scylla-bench with non-shard-aware, the metric will show a lot of cross shard connections from scylla-bench.  

This was a reason why i added to summary info regarding the c-s tool only

File: sdcm/db_stats.py (hunk 2 above)
Comment (aleksbykov) [reply sdcm/db_stats.py:322]: fixed

File: sdcm/tester.py (hunk 1 above)
Comment (aleksbykov) [reply sdcm/tester.py:2927]: removed

File: sdcm/report_templates/results_base.html (hunk 3 above)
Comment (aleksbykov) [reply sdcm/report_templates/results_base.html:209]: will check

Comment (dkropachev) [reply sdcm/report_templates/results_base.html:209]: Yes, we can get it from the same line

**Review by dkropachev — APPROVED** [review]

File: sdcm/report_templates/results_base.html (hunk 3 above)
Comment (fgelcer) [reply sdcm/report_templates/results_base.html:209]: if we can do, @aleksbykov , how hard would be to add this information in this line?

**Review by enaydanov — APPROVED** [review]

**Comment by enaydanov** [comment]
@aleksbykov this PR can't be applied cleanly to branch-4.5. Please, prepare separate PR for this branch.

**Comment by roydahan** [comment]
I don’t think we should backport this to 4.5

**Comment by enaydanov** [comment]
OK

**Comment by aleksbykov** [comment]
@enaydanov @roydahan i think it doesn't need on branch 4.5. I removed the tag

**Pull request merged**